src/tests/stdlib/memcpy.c

```diff
@@ -0,0 +1,75 @@
+/**
+ * @file
+ *
+ * @date Mar 22, 2020
+ * @author: Athreya Nagaraj
+ */
+
+#include <string.h>
+#include <embox/test.h>
+
+EMBOX_TEST_SUITE("memcpy test suit");
+
+TEST_CASE("memcpy with zero size") {
+	char src[] = "string";
+	char dst[sizeof(src) + 1];
+
+	dst[0] = '\0';
+	test_assert_zero(strlen(memcpy(dst, src, 0)));
+}
+
+TEST_CASE("copy string with memcpy") {
+	char src[] = "string";
+	char dst[sizeof(src) + 1];
+
+	memset(dst, 0x55, sizeof(dst));
+
+	test_assert_equal(strlen(src),
+		strlen(memcpy(dst, src, sizeof(dst))));
+	test_assert_zero(strcmp(dst, src));
+}
+
+TEST_CASE("copy integer array with memcpy") {
+	int src[] = {1, 2, 3, 4, 5};
+	int dst[sizeof(src) / sizeof(int)];
+
+	memset(dst, 0x55, sizeof(dst));
+	memcpy(dst, src, sizeof(dst));
+
+	for(int i = 0; i < (sizeof(dst) / sizeof(int)); i++) {
+		test_assert_equal(src[i], dst[i]);
+	}
+}
+
+TEST_CASE("copy char array with null in middle") {
+	char src[] = {'a', 't', 'h', '\0', '-', '9', '0'};
+	char dst[sizeof(src)];
+
+	memset(dst, 0x55, sizeof(dst));
+	memcpy(dst, src, sizeof(dst));
+
+	for(int i = 0; i < sizeof(dst); i++) {
+		test_assert_equal(src[i], dst[i]);
+	}
+}
+
+TEST_CASE("copy within a single array without overflow") {
+	char arr[400];
```

Comment:
I think You are able to just declare two arrays (small and big) and copy the small array to the middle of the big array